store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -567,6 +577,11 @@ public PutMessageResult putMessage(final MessageExtBrokerInner msg) {
         MappedFile unlockMappedFile = null;
         MappedFile mappedFile = this.mappedFileQueue.getLastMappedFile();
 
+        //maybe need to wrap the exception
```

Comment:
When encoding fails, let's notify a user with a `PutMessageResult` about the failure. There's nothing else we can do, I think.